### PR TITLE
Adding groups of paths

### PR DESCRIPTION
**Pull request by dpanta94**
The idea behind this PR is that the existing logic locks us in only using ONE specific root path and ONE relative path. This is not the case for most plugins/themes and much more for the TEC products.

Registering a group of paths can enable us managing multiple locations of assets with the single instance stellar assets.

Use cases:

Declare a group for Event Tickets core
Declare a group for Event Tickets seating
Declare a group for Event Tickets Plus
Declare a group for The events calendar
Declare a group for Events Pro
etc...

### CODE REVIEW

File: tests/wpunit/ConfigTest.php [hunk 1]
```diff
@@ -75,4 +75,43 @@ public function should_reset() {
 			$this->assertInstanceOf( \RuntimeException::class, $e );
 		}
 	}
+
+	/**
+	 * @test
+	 */
+	public function should_add_group_paths() {
+		Config::add_group_path( 'my-group-path-1', [ 'root' => dirname( dirname( __DIR__ ) ) . '/src/feature-1', 'relative' => 'app-1' ] );
```
Comment (lucatume) [inline tests/wpunit/ConfigTest.php:83]: From what I gather, the second parameter of `Config::add_group_path` is an array that has to  define the `root` and `relative` keys. 

Why not change the method signature to something clearer like this?

```php
Config::add_group_path(string $group_slug, string $root, string $relative)
```

**Review by lucatume — CHANGES_REQUESTED** [review]

File: tests/wpunit/ConfigTest.php [hunk 2]
```diff
@@ -75,4 +75,25 @@ public function should_reset() {
 			$this->assertInstanceOf( \RuntimeException::class, $e );
 		}
 	}
+
+	/**
+	 * @test
+	 */
+	public function should_add_group_paths() {
+		Config::add_group_path( 'my-group-path-1', dirname( dirname( __DIR__ ) ) . '/src/feature-1', 'app-1');
```
Comment (lucatume) [inline tests/wpunit/ConfigTest.php:83]:
```suggestion
		Config::add_group_path( 'my-group-path-1', dirname( __DIR__, 2) . '/src/feature-1', 'app-1');
```

File: tests/wpunit/ConfigTest.php [hunk 3]
```diff
@@ -75,4 +75,25 @@ public function should_reset() {
...
+		Config::add_group_path( 'my-group-path-2', dirname( dirname( __DIR__ ) ) . '/src/feature-2', 'app-2');
```
Comment (lucatume) [inline tests/wpunit/ConfigTest.php:84]:
```suggestion
		Config::add_group_path( 'my-group-path-2', dirname( __DIR__, 2) . '/src/feature-2', 'app-2');
```

File: tests/wpunit/AssetsTest.php [hunk 4]
```diff
@@ -120,6 +120,47 @@ public function it_should_get_the_correct_url_when_wp_content_dir_and_wp_content
...
+		Config::set_relative_asset_path( 'tests/_data/' );
+		Config::add_group_path( 'fake-group-path', constant( 'WP_PLUGIN_DIR' ) . '/assets/tests', '_data/fake-feature' );
+
+		foreach ( array_keys( $slugs ) as $slug ) {
+			Asset::add( $slug . '-script', $slug . '.js' )->add_to_group_path( 'fake-group-path' );
+			Asset::add( $slug . '-style', $slug . '.css' )->add_to_group_path( 'fake-group-path' );
+		}
+
+		foreach ( $slugs as $slug => $data ) {
+			$this->assert_minified_found( $slug, true, $data['0'], $data['1'], $id, 'fake-group-path', '/assets/tests/_data/fake-feature/' );
```
Comment (lucatume) [inline tests/wpunit/AssetsTest.php:159]: Use of `$data['0']` and `$data['1']` here looks strange.
Use more ablative names for the data keys?

File: tests/wpunit/AssetsTest.php [hunk 5]
```diff
@@ -120,6 +120,47 @@ public function it_should_get_the_correct_url_when_wp_content_dir_and_wp_content
...
+			$this->assertEquals( $value, constant( $constant ) );
+		}
+
+		Config::reset();
+
+		Config::set_hook_prefix( 'bork' );
+		Config::set_version( '1.1.0' );
+		Config::set_path( constant( 'WP_PLUGIN_DIR' ) . '/assets' );
+		Config::set_relative_asset_path( 'tests/_data/' );
+		Config::add_group_path( 'fake-group-path', constant( 'WP_PLUGIN_DIR' ) . '/assets/tests', '_data/fake-feature' );
```
Comment (lucatume) [inline tests/wpunit/AssetsTest.php:151]: This test uses a group path, but the group path is really an extension of the base one:
* base `constant( 'WP_PLUGIN_DIR' ) . '/assets'`
* group path `constant( 'WP_PLUGIN_DIR' ) . '/assets/tests'`

Can you write another test (do not modify this one that is exercising an edge case), where the group path is NOT an extension of the root path?

**Review by lucatume — CHANGES_REQUESTED** [review]

File: tests/wpunit/AssetsTest.php [hunk 6]
```diff
@@ -108,15 +148,58 @@ public function it_should_get_the_correct_url_when_wp_content_dir_and_wp_content
...
+			$this->assertEquals( $value, constant( $constant ) );
+		}
+
+		Config::reset();
+
+		Config::set_hook_prefix( 'bork' );
+		Config::set_version( '1.1.0' );
+		Config::set_path( constant( 'WP_PLUGIN_DIR' ) . '/assets' );
+		Config::set_relative_asset_path( 'tests/_data/' );
+		// Now are scripts are using a path that does not actually exist in the filesystem. So we can't expect it to figure out minified vs un-minified. So we are adding a new param.
```
Comment (lucatume) [inline tests/wpunit/AssetsTest.php:192]:
```suggestion
		// Now our scripts are using a path that does not actually exist in the filesystem. So we can't expect it to figure out minified vs un-minified. So we are adding a new param.
```

**Review by lucatume — APPROVED** [review]
Smaller typo change, then good for me.
